amethyst_utils/src/ortho_camera.rs

```diff
@@ -5,39 +5,93 @@ use amethyst_core::specs::{
 use amethyst_core::Axis2;
 use amethyst_renderer::{Camera, ScreenDimensions};
 
+/// The coordinates that `OrthoCamera` will keep visible in the window
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OrthoCameraWorldCoordinates {
+    /// Left x coordinate
+    pub left: f32,
+    /// Right x coordinate
+    pub right: f32,
+    /// Bottom y coordinate
+    pub bottom: f32,
+    /// Top y coordinate
+    pub top: f32,
+}
+
+impl OrthoCameraWorldCoordinates {
+    /// Creates coordinates with (0,0) at the bottom left, and (1,1) at the top right
+    pub fn normalized() -> OrthoCameraWorldCoordinates {
+        OrthoCameraWorldCoordinates {
+            left: 0.0,
+            right: 1.0,
+            bottom: 0.0,
+            top: 1.0,
+        }
+    }
+
+    /// width / height of the desired camera coordinates
+    pub fn aspect_ratio(&self) -> f32 {
+        self.width() / self.height()
+    }
+    /// size of the x-axis
+    pub fn width(&self) -> f32 {
+        self.right - self.left
+    }
+    /// size of the y-axis
+    pub fn height(&self) -> f32 {
+        self.top - self.bottom
+    }
+}
+
+impl Default for OrthoCameraWorldCoordinates {
+    fn default() -> Self {
+        Self::normalized()
+    }
+}
+
 /// `Component` attached to the camera's entity that allows automatically adjusting the camera's matrix according
-/// to preferences in the "mode" field.
-/// It tries as much as possible to adjust the camera so that the world's coordinate (0, 0) is at the bottom left and
-/// (1, 1) is at the top right of the window.
-/// You must add the `CameraNormalOrthoSystem` to your dispatcher for this to take effect.
+/// to preferences in the "mode" and "world_coordinates" fields.
+/// It adjusts the camera so that the camera's world coordinates are always visible.
+/// You must add the `CameraNormalOrthoSystem` to your dispatcher for this to take effect (no dependencies required).
 #[derive(Default, Debug, Clone, Serialize, Deserialize)]
-pub struct CameraNormalOrtho {
+pub struct OrthoCamera {
```

Comment:
Please keep Camera at the start of the name. When the feature was first merged in, the reviewers choosed to have Camera at the start of the name to help autocompletion.

Reply:
I don't really agree with that, but I changed it.